deep_water_level/src/infer.py

```diff
@@ -69,116 +71,128 @@ def run_gradio_app(model, crop_box=None):
 
 def run_dataset_inference(
     model,
-    dataset_dir,
-    annotations_file,
+    dataset_dir: Path,
+    annotations_file: str,
     normalized_output,
     crop_box=None,
     use_water_line=False,
 ):
-    # tensor to string
-    def t2s(t):
-        t = t.squeeze()
-        if len(t.shape) == 0 or t.shape[0] == 1:
-            return f"{t.item():.2f}"
-
-        return "[" + ", ".join([f"{x:.2f}" for x in t]) + "]"
-
-    # Load the dataset
-    dataset = get_data_loader(
-        dataset_dir + "/images",
-        dataset_dir + "/annotations/" + annotations_file,
-        shuffle=False,
-        crop_box=crop_box,
+    # Convert numpy array to string for printing
+    def a2s(a):
+        assert len(a.shape) <= 1
+        if len(a.shape) == 0 or a.shape[0] == 1:
+            return f"{a.item():.4f}"
+        else:
+            return "[" + ", ".join([f"{x:.4f}" for x in a]) + "]"
+
+    transforms = get_transforms(crop_box=crop_box)
+    dataset = WaterDataset(
+        dataset_dir / "annotations" / annotations_file,
+        dataset_dir / "images",
+        transforms=transforms,
         normalize_output=normalized_output,
```

Comment:
I think normalize_output is not used anymore in WaterDataset, in that case it should be removed

Reply:
Agreed - I left it in this PR but we can remove it in another.